GVFS/GVFS.Service/Mac/GVFSService.cs

```diff
@@ -0,0 +1,115 @@
+using GVFS.Common;
+using GVFS.Common.NamedPipes;
+using GVFS.Common.Tracing;
+using System;
+using System.Linq;
+using System.Threading;
+
+namespace GVFS.Service
+{
+    public class GVFSService
+    {
+        private const string ServiceNameArgPrefix = "--servicename=";
+        private const string EtwArea = nameof(GVFSService);
+
+        private JsonTracer tracer;
+        private Thread serviceThread;
+        private ManualResetEvent serviceStopped;
+        private string serviceName;
+        private string serviceDataLocation;
+
+        public GVFSService(JsonTracer tracer)
+        {
+            this.tracer = tracer;
+            this.serviceName = GVFSConstants.Service.ServiceName;
+        }
+
+        public void RunWithArgs(string[] args)
+        {
+            if (this.serviceThread != null)
+            {
+                throw new InvalidOperationException("Cannot start service twice in a row.");
+            }
+
+            string nameArg = args.FirstOrDefault(arg => arg.StartsWith(ServiceNameArgPrefix, StringComparison.OrdinalIgnoreCase));
+            if (nameArg != null)
+            {
+                this.serviceName = nameArg.Substring(ServiceNameArgPrefix.Length);
+            }
+
+            try
+            {
+                this.serviceDataLocation = Paths.GetServiceDataRoot(this.serviceName);

```

Comment:
What will `Paths.GetServiceDataRoot(this.serviceName)` be on Mac?

Reply:
it returns `/usr/share/GVFS/GVFS.Service`